wordpress-mobile/WordPress-Android
# My Site Dashboard: Tabs - Refactor My Site Header

Fixes #16031 

This PR refactors My Site Header to the App bar.  

Following are the Screenshots for the same functionality. 

| Site header when tabs enabled  | Site header when tabs disabled | 
|---|---|
| ![Screenshot_20220323_110733](https://user-images.githubusercontent.com/17463767/159645770-5f3ad934-4224-4e81-bd0d-334a6bed131f.png)| ![Screenshot_20220321_233806](https://user-images.githubusercontent.com/17463767/159645796-390c5454-1c56-4f1b-b020-d128a8bbd074.png)  |   |   |   |
| On Collapsed(Scrolled to bottom)  |   | 
| ![Screenshot_20220321_233358](https://user-images.githubusercontent.com/17463767/159645852-a2f1399a-a5f3-474b-8be0-b6338356fda9.png) |  ![Screenshot_20220321_233757](https://user-images.githubusercontent.com/17463767/159645910-d143a5bb-2626-4385-80b7-b3c5e588d68b.png) | 

### Known issues and Enhancements (Tracked in #16002)
---
Enhancements 
- #16153 

Issues
- #16154
- #16155
- #16156


##  Merge Instructions  🟢 
Merge only once the below issues are fixed
- #16153 
- #16156


### To test:
---
Test 1
- Launch the app.
- Go to My Site
- Notice that the Site Info is shown above the quick links and collapses to Site title when scrolled below 
- All Interactions on the Site Info card(title,Url,Icon and Site Selector Navigation) works as expected 

Test 2 
- Launch the app.
- Go to My Site -> Me -> App Settings -> Debug settings.
- Turn on the MySiteDashboardTabsFeatureConfig flag under the Features in development section.
- Restart the app
- Notice that the Site Info is shown above the Tabs and collapses to Site title when scrolled below 
- All Interactions on the Site Info card(title,Url,Icon and Site Selector Navigation) works as expected 

## Regression Notes
1. Potential unintended areas of impact
- My Site Dashboard not being shown properly
- Site Info not being shown 


2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual tests and Unit tests

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
